ibm_mq/datadog_checks/ibm_mq/connection.py

```diff
@@ -51,10 +51,10 @@ def get_ssl_connection(config):
     Get the connection with SSL
     """
     cd = _get_channel_definition(config)
-    cd.SSLCipherSpec = config.ssl_cipher_spec
+    cd.SSLCipherSpec = pymqi.ensure_bytes(config.ssl_cipher_spec)
```

Comment:
There is an `ensure_bytes()` in `datadog_checks.base`, any pointers on the trade-off between that one and pymqi's? I assume they do the same thing, but I'm not able to find `ensure_bytes()` in the pymqi docs. Should we use our own?

Reply:
This is pymqi implementation:
```
def ensure_bytes(s, encoding='ascii'):
    if is_unicode(s):
        return s.encode(encoding)
    else:
        return s
```
This is checks_base implementation:
```
def ensure_bytes(s):
    # type: (Union[Text, bytes]) -> bytes
    if isinstance(s, text_type):
        s = s.encode('utf-8')
    return s
```
They default to different encoding

Reply:
> There is an `ensure_bytes()` in `datadog_checks.base`, any pointers on the trade-off between that one and pymqi's? 

There are 2 differences:
- `pymqi.ensure_bytes` won't convert Python2 convert python unicode `u'abc'`. I think it's mainly intended for Python3
- the default encoding is `ascii`

I think both differences doesn't make difference in practice since we don't pass Python 2 unicode strings and the arguments are only ascii chars.

Details:

```
def is_unicode(s):
    """ Returns True if input arg is a Python 3 string (aka Python 2 unicode). False otherwise.
    """
    if isinstance(s, str) and not isinstance(s, bytes):
        return True
    else:
        return False

def ensure_bytes(s, encoding='ascii'):
    print("is_unicode(s)", is_unicode(s))
    if is_unicode(s):
        return s.encode(encoding)
    else:
        return s


assert isinstance(ensure_bytes('abc'), bytes)      # ok for py2 and py3
assert isinstance(ensure_bytes(b'abc'), bytes)    # ok for py2 and py3
assert isinstance(ensure_bytes(u'abc'), bytes)    # this fail for Python 2
```




```
from six import text_type

def datadog_checks_base_ensure_bytes(s):
    # type: (Union[Text, bytes]) -> bytes
    if isinstance(s, text_type):
        s = s.encode('utf-8')
    return s


assert isinstance(datadog_checks_base_ensure_bytes('abc'), bytes) # ok for py2 and py3
assert isinstance(datadog_checks_base_ensure_bytes(b'abc'), bytes) # ok for py2 and py3
assert isinstance(datadog_checks_base_ensure_bytes(u'abc'), bytes) # ok for py2 and py3
```

> I assume they do the same thing, but I'm not able to find `ensure_bytes()` in the pymqi docs. Should we use our own?

I tend to prefer to use pymqi version for consistency with the library internal use of `ensure_bytes` and in their tests.

It's not documented, but I think the risk is low that it's removed. Even if it happen, we can still address the issue quite easily.

We currently already use it here: https://github.com/DataDog/integrations-core/blob/826f63c105410c650b5a0b8f2a1a2fb520da3952/ibm_mq/datadog_checks/ibm_mq/connection.py#L68-L69

Reply:
Thanks for the very detailed explanations 😄